Fixed issue #471  Forgot password using Firebase Authentication

nakul30 opened the pull request:
This pull request resolves the issue no #471 and implements forgot password using firebase authentication

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Implemented a "Forgot Password" feature, allowing users to reset their password through email.

- **Enhancements**
  - Added a "Forgot Password" link to the Login component for easier navigation.
  - Updated the Signup component to utilize a `Link` component for the "forgot password" functionality.

- **Security**
  - Enhanced security by adding `rel="noreferrer"` to social media links in user profiles.

- **Refactor**
  - Standardized the use of quotes in the `useMediaQuery` function call.
  - Improved code readability in backend function declarations by adjusting spacing.

- **Documentation**
  - Updated server configuration documentation with new host settings and React plugin integration.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

anur4ag commented:
Hey @nakul30 the issue seems to be fixed , however can we also change the styling of the password reset page or is it default by firebase ?

nakul30 commented:
> Hey @nakul30 the issue seems to be fixed , however can we also change the styling of the password reset page or is it default by firebase ?

Its by default by the firebase

anur4ag commented:
> > Hey @nakul30 the issue seems to be fixed , however can we also change the styling of the password reset page or is it default by firebase ?
> 
> Its by default by the firebase

Ohk no problems, but make sure that `localhost:5173/forgot-password` route automatically redirects to `/u/dashboard` if a user is already logged in, like currently if a user is logged in and he/she goes to `/login` then they are automatically redirected to `/u/dashboard`

nakul30 commented:
> > > Hey @nakul30 the issue seems to be fixed , however can we also change the styling of the password reset page or is it default by firebase ?
> > 
> > 
> > Its by default by the firebase
> 
> Ohk no problems, but make sure that `localhost:5173/forgot-password` route automatically redirects to `/u/dashboard` if the _user is already logged in_, like currently if a user is logged in and he/she goes to `/login` then they are automatically redirected to `/u/dashboard`

I have just fixed this issue in my new Commit. I have updated the code and now it will redirect to dashboard if user is already logged in

anur4ag commented:
Great!, just one more change use firebase to `pass state in action email` and redirect the user back to login page once he has reset his password using the email link.

For more information read - [https://firebase.google.com/docs/auth/web/passing-state-in-email-actions]

nakul30 commented:
> Great!, just one more change use firebase to `pass state in action email` and redirect the user back to login page once he has reset his password using the email link.
> 
> For more information read - [https://firebase.google.com/docs/auth/web/passing-state-in-email-actions]

I have written the code in way that login button appears after the link is sent so in case the wrong email address is filled user not have to shift the pages

anur4ag commented:
After the user has successfully reset the password on fireabase's page the current workflow looks like this :
reset password mail -> link click -> new password enter -> password sucessfully reset -> close that tab -> then go to digitomize.com/login 

What I meant after the following tab appears, fireabase should automatically redirect the user to `/login` , without him closing this tab then going to login himself

<img width="400" alt="Screenshot 2024-01-02 at 14 33 07" src="https://github.com/digitomize/digitomize/assets/71564387/324469c3-aeb1-43ee-9f17-2a39210f2b6f">

nakul30 commented:
> After the user has successfully reset the password on fireabase's page the current workflow looks like this : reset password mail -> link click -> new password enter -> password sucessfully reset -> close that tab -> then go to digitomize.com/login
> 
> What I meant was after the following tab appears, fireabase should automatically redirect the user to `/login` , without him closing this tab then going to login himself
> 
> <img alt="Screenshot 2024-01-02 at 14 33 07" width="400" src="https://private-user-images.githubusercontent.com/71564387/293658060-324469c3-aeb1-43ee-9f17-2a39210f2b6f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQxODY3OTIsIm5iZiI6MTcwNDE4NjQ5MiwicGF0aCI6Ii83MTU2NDM4Ny8yOTM2NTgwNjAtMzI0NDY5YzMtYWViMS00M2VlLTlmMTctMmEzOTIxMGYyYjZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAyVDA5MDgxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjMmEzZDA0YjAzYWMzY2I2YTIxMjA3NTkxNGQzNjAwMDdjYzAyYjJmMTY4NzgwM2M3OTVkMTViZGU5ODlmNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8_LJhx5y-KHpM_uMNk8xDFBF3PMUbKRg8mYZWc4hEx4">
> It can be achieved through the docs I sent before

Alright Anurag, I an just working on it

nakul30 commented:
> > After the user has successfully reset the password on fireabase's page the current workflow looks like this : reset password mail -> link click -> new password enter -> password sucessfully reset -> close that tab -> then go to digitomize.com/login
> > What I meant was after the following tab appears, fireabase should automatically redirect the user to `/login` , without him closing this tab then going to login himself
> > <img alt="Screenshot 2024-01-02 at 14 33 07" width="400" src="https://private-user-images.githubusercontent.com/71564387/293658060-324469c3-aeb1-43ee-9f17-2a39210f2b6f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQxODY3OTIsIm5iZiI6MTcwNDE4NjQ5MiwicGF0aCI6Ii83MTU2NDM4Ny8yOTM2NTgwNjAtMzI0NDY5YzMtYWViMS00M2VlLTlmMTctMmEzOTIxMGYyYjZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAyVDA5MDgxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjMmEzZDA0YjAzYWMzY2I2YTIxMjA3NTkxNGQzNjAwMDdjYzAyYjJmMTY4NzgwM2M3OTVkMTViZGU5ODlmNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8_LJhx5y-KHpM_uMNk8xDFBF3PMUbKRg8mYZWc4hEx4">
> > It can be achieved through the docs I sent before
> 
> Alright Anurag, I an just working on it

I am unable to import ActionCodeSettings in my component as it is leading to Appcrash

nakul30 commented:
Check this Anurag i think its alright now

anur4ag commented:
> Check this Anurag its alright now

Do I need to add anything in my firebase app console as it is giving me this error - `Firebase: Domain not whitelisted by project (auth/unauthorized-continue-uri).`

nakul30 commented:
> > Check this Anurag its alright now
> 
> Do I need to add anything in my firebase app console as it is giving me this error - `Firebase: Domain not whitelisted by project (auth/unauthorized-continue-uri).` ?

you have to put the login domain in authorized domains of the app in the firebase

anur4ag commented:
Great work @nakul30 , that's all we needed for the complete working of the forgot-password issue. I will merge this PR.

Please send the email template and how to set in firebase ?

nakul30 commented:
> Great work @nakul30 🙌 , that's all we needed for the complete working of the forgot-password issue. I will merge this PR.
> 
> Please send the email template and how to set in firebase ?

You can find the template in Template section of Authentication window in Firebase.
There you can set custom actionurl too.

anur4ag commented:
@nakul30 I've added you as a contributor in our repository's readme. Keep contributing .
Thanks! 😄
